include/envoy/http/access_log.h

```diff
@@ -89,9 +91,10 @@ class RequestInfo {
   virtual std::chrono::milliseconds duration() const PURE;
 
   /**
-   * @return the failure reason for richer log experience.
+   * @return the failure reason for richer log experience, this can be represented as binary OR of
+   * FailureReason enum values.
    */
-  virtual FailureReason failureReason() const PURE;
+  virtual uint64_t failureReason() const PURE;
```

Comment:
i assume that if we change FailureReason -> ResponseFlag we'll also need to change access log formatting to allow specifying %RESPONSE_FLAG% instead of %FAILURE_REASON% which will not be backward compatible. Any concerns around this at this point? (If not then I'll go for ReponseFlag).